src/main/java/uk/gov/digital/justice/hmpps/sentenceplan/service/SentencePlanService.java

```diff
@@ -132,35 +131,36 @@ public void updateMotivations(UUID sentencePlanUuid, Map<UUID, UUID> newMotivati
     }
 
     @Transactional
-    public void updateActionPriorities(UUID sentencePlanUuid, Map<UUID, Integer> priorities) {
-        if (priorities.size() > 0) {
+    public void updateActionPriorities(UUID sentencePlanUuid, Map<UUID, Integer> newPriorities) {
+        if (newPriorities.size() > 0) {
 
             // Map to a set to get a unique set of values
-            Set<Integer> uniqueValues = new HashSet<>(priorities.values());
-            if (uniqueValues.size() < priorities.size()) {
+            Set<Integer> uniqueValues = new HashSet<>(newPriorities.values());
+            if (uniqueValues.size() < newPriorities.size()) {
                 throw new ValidationException("actions with duplicate priority found.");
             }
 
             var sentencePlan = getSentencePlanEntity(sentencePlanUuid);
             var planActions = sentencePlan.getData().getActions().stream().collect(Collectors.toMap(ActionEntity::getId, action -> action));
 
-            // We also need to check that we're updating all the actions otherwise they will get out of action (no pun intended).
-            if (planActions.size() != priorities.size()) {
+            // We also need to check that we're updating all the actions otherwise they will get out of step.
```

Comment:
Are you sure about this? You could update only the ones you get sent and leave the remaining actions unchanged?